### ISSUE TITLE
Different results with gometalinter

### ISSUE THREAD

**Issue by schroding3rscat** [golangci/golangci-lint#231]
Hi,
I tried to use `golangci-lint` and it's very fast. But I often have an issue with different results from `gometalinter`. `golangci-lint` almost always produces fewer warnings with the same linters enabled. I saw this behavior not only for `ineffassign` which example I attached at the bottom but with other linters too. Why does this happen? How can I fix this? Right now `golangci-lint` looks unreliable for us.

1. Version of golangci-lint: 
```
C:\Users\user\Downloads
λ go get -u github.com/golangci/golangci-lint/cmd/golangci-lint

C:\Users\user\Downloads
λ golangci-lint --version
Error: unknown flag: --version
Usage:
  golangci-lint [flags]
  golangci-lint [command]

Available Commands:
  help        Help
  help        Help about any command
  linters     List current linters configuration
  run         Run this tool in cloud on every github pull request in https://golangci.com for free (public repos)

Flags:
  -j, --concurrency int           Concurrency (default NumCPU) (default 4)
      --cpu-profile-path string   Path to CPU profile output file
  -h, --help                      help for golangci-lint
      --mem-profile-path string   Path to memory profile output file
  -v, --verbose                   verbose output

Use "golangci-lint [command] --help" for more information about a command.

panic: unknown flag: --version

goroutine 1 [running]:
main.main()
        C:/git/go/src/github.com/golangci/golangci-lint/cmd/golangci-lint/main.go:17 +0xdb
```
2. Go environment: 
```
λ go version && go env
go version go1.10.4 windows/amd64
set GOARCH=amd64
set GOBIN=
set GOCACHE=C:\Users\user\AppData\Local\go-build
set GOEXE=.exe
set GOHOSTARCH=amd64
set GOHOSTOS=windows
set GOOS=windows
set GOPATH=C:\git\go
set GORACE=
set GOROOT=C:\Go
set GOTMPDIR=
set GOTOOLDIR=C:\Go\pkg\tool\windows_amd64
set GCCGO=gccgo
set CC=gcc
set CXX=g++
set CGO_ENABLED=1
set CGO_CFLAGS=-g -O2
set CGO_CPPFLAGS=
set CGO_CXXFLAGS=-g -O2
set CGO_FFLAGS=-g -O2
set CGO_LDFLAGS=-g -O2
set PKG_CONFIG=pkg-config
set GOGCCFLAGS=-m64 -mthreads -fmessage-length=0 -fdebug-prefix-map=C:\Users\user\AppData\Local\Temp\go-build648297742=/tmp/go-build -gno-record-gcc-switches
```
3. Output of running linters:
```
C:\git\go\src\git.company.com\ci\service (feature/add-tests -> origin)
λ golangci-lint.exe run --disable-all --enable=ineffassign ./...                                                       
                                                                                                                       
C:\git\go\src\git.company.com\ci\service (feature/add-tests -> origin)
λ gometalinter --disable-all --enable=ineffassign --vendor ./...                                                       
test\account\account.go:220:7:warning: ineffectual assignment to err (ineffassign)                               
test\account\account.go:255:7:warning: ineffectual assignment to err (ineffassign)                               
test\account\account.go:290:7:warning: ineffectual assignment to err (ineffassign)                               
```

Code from account.go:220 below. As you can see, the warning is right. I have `req, err = ` and this value assigned to `err` wasn't used anywhere.
```
var req *http.Request
    req, err = http.NewRequest("POST", apiURL, bytes.NewBuffer(j))
    req.Header.Add("Authorization", accessToken)
    req.Header.Add("Content-Type", "application/json")

    resp, err := client.Do(req)
    if err != nil {
        return
    }
```

Here's a verbose output from `golangci-lint` (I stripped some values to not to share our code):
```
λ golangci-lint.exe run --disable-all --enable=ineffassign ./... -v
level=info msg="[config_reader] Config search paths: [./ C:\\git\\go\\src\\git.company.com\\ci\\service C:\\git\\go\\src\\git.company.com\\ci C:\\git\\go\\src\\git.company.com C:\\git\\go\\src C:\\git\\go C:\\git C:\\]"
level=info msg="[lintersdb] Active 1 linters: [ineffassign]"
level=info msg="[load/path_resolver] Skipped dirs: [.git .vs .vscode deploy\\msp\\roles\\_legacy deployments\\msp\\roles\\_legacy vendor]"
level=info msg="[load/path_resolver] Paths resolving took 3.9590543s: [cmd\\config-gen cmd\\service pkg\\server test\\account]"
level=info msg="[runner] worker.4 took 0s"
level=info msg="[runner] worker.1 took 0s"
level=info msg="[runner] worker.3 took 0s"
level=info msg="[runner] worker.2 took 46.9987ms with stages: ineffassign: 46.9987ms"
level=info msg="[runner] Workers idle times: #1: 34.988ms, #3: 29.9964ms, #4: 41.9951ms"
level=info msg="[runner] processing took 0s with stages: skip_files: 0s, autogenerated_exclude: 0s, exclude: 0s, max_per_file_from_linter: 0s, max_same_issues: 0s, max_from_linter: 0s, path_prettifier: 0s, cgo: 0s, diff: 0s, source_code: 0s, nolint: 0s, uniq_by_line: 0s"
level=info msg="Memory: 42 samples, avg is 21.1MB, max is 24.0MB"
level=info msg="Execution took 4.1550773s"
```

**Comment by vetheslav** [comment]
I have the same issue with the latest version on golangci and with v1.10+ versions

**Comment by jirfag** [comment]
thank you for reporting!
we already have autotests for it. Also I've manually tested this snippet:
```go
package testdata

import (
        "bytes"
        "fmt"
        "net/http"
)

func ineff() {
        var req *http.Request
        var err error
        req, err = http.NewRequest("POST", "apiURL", bytes.NewBuffer([]byte("data")))
        req.Header.Add("Authorization", "accessToken")
        req.Header.Add("Content-Type", "application/json")

        resp, err := http.DefaultClient.Do(req)
        if err != nil {
                return
        }

        fmt.Print(resp)
}
```

on version 1.10.2 and 1.11.1:
```sh
~/go/src/github.com/golangci/golangci-lint ((v1.10.2) *)$ go run ./cmd/golangci-lint/main.go run ./test/testdata/ineffassign.go --no-config --disable-all -Eineffassign
test/testdata/ineffassign.go:14:3: ineffectual assignment to `x` (ineffassign)
                x = 0 // ERROR "ineffectual assignment to `x`"
                ^
test/testdata/ineffassign.go:22:7: ineffectual assignment to `err` (ineffassign)
        req, err = http.NewRequest("POST", "apiURL", bytes.NewBuffer([]byte("data")))
             ^
exit status 1
```

Both versions work ok.
Maybe it's a windows issue, maybe some details in your code or config make a difference. I've seen from `-v` output that you have a lot of skipped dirs, maybe this file was in a such skipped dir? Or maybe this file was marked as autogenerated.

Can you make a minimally reproducible example (including code and configuration)?
